### PR TITLE
Request signature optional

### PR DESCRIPTION
**Pull request by marekrjpolak**
Skips `mask_public_key` and `signature` checks when they're received as empty from client.

Legacy fw changes were difficult to test because of [this issue](https://github.com/trezor/trezor-firmware/issues/3892), but the same changes applied to 1.12.1 worked fine.

More rationale in Notion: https://www.notion.so/satoshilabs/Request-signature-checks-removal-bc5f3174ca9543e6ae31b6bc08087e76

### CODE REVIEW

**Review by M1nd3r — APPROVED** [review]
lgtm

File: core/src/apps/bitcoin/sign_tx/approvers.py [hunk 1]
```diff
@@ -435,46 +430,17 @@ def __init__(
...
-        writers.write_uint64(self.h_request, self.request.min_registrable_amount)
-        writers.write_bytes_fixed(self.h_request, self.request.mask_public_key, 33)
-        writers.write_compact_size(self.h_request, tx.inputs_count)
-
         # Upper bound on the user's contribution to the weight of the transaction.
         self.our_weight = tx_weight.TxWeightCalculator()
 
     def _add_input(self, txi: TxInput) -> None:
         super()._add_input(txi)
-        writers.write_uint8(self.h_request, txi.coinjoin_flags)
```
Comment (matejcik) [inline core/src/apps/bitcoin/sign_tx/approvers.py:455]: pretty sure you can drop the whole function

File: tests/device_tests/bitcoin/payment_req.py [hunk 2]
```diff
@@ -113,60 +113,15 @@ def make_coinjoin_request(
...
-    h_request.update(h_outputs.digest())
-
     return messages.CoinJoinRequest(
         fee_rate=fee_rate,
         no_fee_threshold=no_fee_threshold,
         min_registrable_amount=min_registrable_amount,
-        mask_public_key=mask_public_key,
-        signature=payment_req_signer.sign_digest_deterministic(h_request.digest()),
+        mask_public_key=bytearray(),
+        signature=bytearray(),
```
Comment (matejcik) [inline tests/device_tests/bitcoin/payment_req.py:126]:
```suggestion
        mask_public_key=b"",
        signature=b"",
```

Comment (marekrjpolak) [reply tests/device_tests/bitcoin/payment_req.py]: https://github.com/trezor/trezor-firmware/pull/3891/commits/36ea137cd8909f4529c526dd89e61153ce6b3e0e

**Pull request merged**